exporter/splunkhecexporter/client.go

```diff
@@ -148,19 +152,150 @@ func (c *client) sendSplunkEvents(ctx context.Context, splunkEvents []*splunk.Ev
 	return nil
 }
 
-func (c *client) pushLogData(ctx context.Context, ld pdata.Logs) error {
+func subLogs(ld *pdata.Logs, start *logIndex) *pdata.Logs {
+	if ld == nil || start == nil || start.zero() {
+		return ld
+	}
+
+	logs := pdata.NewLogs()
+	RL, RL2 := ld.ResourceLogs(), logs.ResourceLogs()
+
+	for r := start.resource; r < RL.Len(); r++ {
+		RL2.Append(pdata.NewResourceLogs())
+		RL.At(r).Resource().CopyTo(RL2.At(r - start.resource).Resource())
+
+		IL, IL2 := RL.At(r).InstrumentationLibraryLogs(), RL2.At(r-start.resource).InstrumentationLibraryLogs()
+
+		i := 0
+		if r == start.resource {
+			i = start.library
+		}
+		for i2 := 0; i < IL.Len(); i++ {
+			IL2.Append(pdata.NewInstrumentationLibraryLogs())
+			IL.At(i).InstrumentationLibrary().CopyTo(IL2.At(i2).InstrumentationLibrary())
+
+			LR, LR2 := IL.At(i).Logs(), IL2.At(i2).Logs()
+			i2++
+
+			l := 0
+			if r == start.resource && i == start.library {
+				l = start.record
+			}
+			for l2 := 0; l < LR.Len(); l++ {
+				LR2.Append(pdata.NewLogRecord())
+				LR.At(l).CopyTo(LR2.At(l2))
+				l2++
+			}
+		}
+	}
+
+	return &logs
+}
+
+func (c *client) sentLogBatch(ctx context.Context, ld pdata.Logs, send func(context.Context, *bytes.Buffer) error) (err error) {
```

Comment:
I have a hard time understanding what this function does. It would be good to add some comments to explain what is going on here. It would be also probably easier to read if it was broken down to smaller functions.  Also describe parameters.